File: sohee-jung/week3/백준_silver_촌수계산.java
import java.io.*;
import java.util.ArrayList;
import java.util.Collections;
import java.util.StringTokenizer;

public class Main {
    static ArrayList<Integer>[] arrayList;
    static boolean[] visitedDfs;
    static int answer = -1;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());
        int firstPerson = Integer.parseInt(st.nextToken());
        int secondPerson = Integer.parseInt(st.nextToken());

        int m = Integer.parseInt(br.readLine());

        arrayList = new ArrayList[n + 1];
        for (int i = 0; i <= n; i++) {
            arrayList[i] = new ArrayList<>();
        }

        visitedDfs = new boolean[n + 1];

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());

            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            arrayList[x].add(y);
            arrayList[y].add(x);
        }

        for (int i = 0; i <= n; i++) {
            Collections.sort(arrayList[i]);
        }

        dfs(firstPerson, secondPerson, 0);
        bw.write(answer+ "\n");

        bw.flush();
        bw.close();
        br.close();
    }

    private static void dfs(int current, int target, int depth) {
        if (current == target) {
            answer = depth;
            return;
        }
        visitedDfs[current] = true;

        for (int next : arrayList[current]) {
            if (!visitedDfs[next]) {
                dfs(next, target, depth + 1);
            }
        }
    }
}

File: minseo-song/week3/백준_silver_촌수계산.kt
package week3

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.LinkedList
import java.util.Queue
import java.util.StringTokenizer

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val n = StringTokenizer(bufferedReader.readLine()).nextToken().toInt()

    val st = StringTokenizer(bufferedReader.readLine())
    val a = st.nextToken().toInt()
    val b = st.nextToken().toInt()

    val m = StringTokenizer(bufferedReader.readLine()).nextToken().toInt()

    val graph = Array(n + 1) { mutableListOf<Int>() }

    repeat(m) {
        val edge = StringTokenizer(bufferedReader.readLine())
        val x = edge.nextToken().toInt()
        val y = edge.nextToken().toInt()

        graph[x].add(y)
        graph[y].add(x)
    }

    val visited = BooleanArray(n + 1)
    val queue: Queue<Pair<Int, Int>> = LinkedList()

    queue.add(Pair(a, 0))
    visited[a] = true

    var result = -1

    while (queue.isNotEmpty()){
        val (current, count) = queue.poll()

        if (current == b){
            result = count
            break
        }

        for (next in graph[current]){
            if (!visited[next]){
                visited[next] = true
                queue.add(Pair(next, count + 1))
            }
        }
    }

    bufferedWriter.write("$result")
    bufferedWriter.close()
}